pinot-segment-local/src/main/java/org/apache/pinot/segment/local/data/manager/TableDataManager.java

```diff
@@ -146,4 +156,12 @@ void addSegment(String segmentName, TableConfig tableConfig, IndexLoadingConfig
    * @return List of {@link SegmentErrorInfo}
    */
   Map<String, SegmentErrorInfo> getSegmentErrors();
+
+  /**
+   * Get the dir for the given table segment, when TableDataManager object is not
+   * yet initialized for the given table.
+   */
+  static File getSegmentDataDir(String instanceDataDir, String tableNameWithType, String segmentName) {
```

Comment:
This is instance level method, and seems only used in `InstanceDataManager`. Can we inline it instead of putting it as an interface method?

Reply:
I was thinking to encapsulate the details of assembling the path to segment under table mgr iface. not feel strong about this though.